editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search.tsx

```diff
@@ -1,132 +1,283 @@
+  display: "block"
+}));
+
+const ExternalPortalCard = styled(ListItemButton)(({ theme }) => ({
+  backgroundColor: "black",
+  color: theme.palette.common.white,
+  "&:hover": {
+    backgroundColor: "black",
+    "& p:last-of-type": {
+      textDecoration: "underline",
+    }
+  }
+}));
+
+const SearchResults: React.FC<{ results: SearchResults<IndexedNode> }> = ({
+  results,
+}) => {
+  return (
+    <>
+      <Typography variant="h3" mb={1}>
+        {!results.length && "No matches found"}
+        {results.length === 1 && "1 result:"}
+        {results.length > 1 && `${results.length} results:`}
+      </Typography>
+
+      <SearchResultRoot>
+        {results.map((result) => (
+          <ListItem key={result.item.id} disablePadding>
+            <SearchResultCard result={result} />
+          </ListItem>
+        ))}
+      </SearchResultRoot>
+    </>
+  );
+};
+
+interface HeadlineProps {
   text: string;
-  path: string[];
+  matchIndices: [number, number][];
+  variant: "data";
 }
 
-const SearchResults: React.FC<{ results: SearchResult[] }> = ({ results }) => {
+const Headline: React.FC<HeadlineProps> = ({ text, matchIndices, variant }) => {
+  const isHighlighted = (index: number) =>
+    matchIndices.some(([start, end]) => index >= start && index <= end);
+
   return (
-    <Box
-      sx={{ width: "100%", gap: 2, display: "flex", flexDirection: "column" }}
-    >
-      {results.map((result) => (
-        <SearchResultCard key={result.nodeId} {...result} />
+    <>
+      {text.split("").map((char, index) => (
+        <Typography
+          fontWeight={isHighlighted(index) ? FONT_WEIGHT_BOLD : "regular"}
+          component="span"
+          p={0}
+          key={`headline-character-${index}`}
+          fontFamily={
+            variant === "data" ? '"Source Code Pro", monospace' : "inherit"
+          }
+          variant="body2"
+          display={"inline-block"}
+        >
+          {char}
+        </Typography>
       ))}
-    </Box>
+    </>
   );
 };
 
-const SearchResultCard: React.FC<SearchResult> = ({
-  nodeTitle,
-  text,
-  nodeType,
+const SearchResultCard: React.FC<{ result: SearchResult<IndexedNode> }> = ({
+  result,
 }) => {
-  const Icon = ICONS[nodeType];
+  const getDisplayDetailsForResult = ({
+    item,
+    key,
+  }: SearchResult<IndexedNode>) => {
+    const componentType = capitalize(
+      SLUGS[result.item.type].replaceAll("-", " "),
+    );
+    let title = (item.data?.title as string) || (item.data?.text as string);
+    let Icon = ICONS[item.type];
+    // TODO: Generate display key from key
+    let displayKey = "Data";
+    const headline = get(item, key).toString() || "";
+
+    // For Answer nodes, update display values to match the parent question
+    if (item.type === ComponentType.Answer) {
+      const parentNode = useStore.getState().flow[item.parentId];
+      Icon = ICONS[ComponentType.Question];
+      title = parentNode!.data.text!;
+      displayKey = "Answer (data)";
```

Comment:
nit: in the Editor modals, we refer to them as "Options" not "Answers" - might be nice to keep user-facing language similar here!